ding/policy/ppo.py

```diff
@@ -954,3 +954,277 @@ def _monitor_vars_learn(self) -> List[str]:
+            heads=[1, 1],
+            # (str) the contrastive loss type.
+            loss_type='infonce',
+            # (float) a parameter to adjust the polarity between positive and negative samples.
+            temperature=1.0,
+        ),
+        learn=dict(
+            # (bool) Whether to use multi gpu
+            multi_gpu=False,
+            epoch_per_collect=10,
+            batch_size=64,
+            learning_rate=3e-4,
+            # ==============================================================
+            # The following configs is algorithm-specific
+            # ==============================================================
+            # (float) The loss weight of value network, policy network weight is set to 1
+            value_weight=0.5,
+            # (float) The loss weight of entropy regularization, policy network weight is set to 1
+            entropy_weight=0.0,
+            # (float) PPO clip ratio, defaults to 0.2
+            clip_ratio=0.2,
+            # (bool) Whether to use advantage norm in a whole training batch
+            adv_norm=True,
+            value_norm=True,
+            ppo_param_init=True,
+            grad_clip_type='clip_norm',
+            grad_clip_value=0.5,
+            ignore_done=False,
+        ),
+        collect=dict(
+            # (int) Only one of [n_sample, n_episode] shoule be set
+            # n_sample=64,
+            # (int) Cut trajectories into pieces with length "unroll_len".
+            unroll_len=1,
+            # ==============================================================
+            # The following configs is algorithm-specific
+            # ==============================================================
+            # (float) Reward's future discount factor, aka. gamma.
+            discount_factor=0.99,
+            # (float) GAE lambda factor for the balance of bias and variance(1-step td and mc)
+            gae_lambda=0.95,
+        ),
+        eval=dict(),
+        aux_loss_weight=0.001,
+    )
+
+    def _init_learn(self) -> None:
+        """
+        Overview:
+            Learn mode init method. Called by ``self.__init__``.
+            Init the auxiliary model, its optimizer, and the axuliary loss weight to the main loss.
+        """
+        super()._init_learn()
+        x_size, y_size = self._get_encoding_size()
+        self._aux_model = ContrastiveLoss(x_size, y_size, **self._cfg.aux_model)
+        if self._cuda:
+            self._aux_model.cuda()
+        self._aux_optimizer = Adam(self._aux_model.parameters(), lr=self._cfg.learn.learning_rate)
+        self._aux_loss_weight = self._cfg.aux_loss_weight
+
+    def _get_encoding_size(self):
+        """
+        Overview:
+            Get the input encoding size of the ST-DIM axuiliary model.
+        Returns:
+            - info_dict (:obj:`Tuple[Tuple]`): the encoding size without the first (Batch) dimension.
```

Comment:
- `Tuple[Tuple, Tuple]`?
- use uppercase letter for comment, i.e., `The encoding size`